services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js

```diff
@@ -164,27 +171,88 @@ export class NOWPermitGeneration extends Component {
     permitGenObject.regional_office = !amendment.regional_office
       ? regionHash[noticeOfWork.mine_region]
       : amendment.regional_office;
+    permitGenObject.now_tracking_number = noticeOfWork.now_tracking_number;
+    permitGenObject.now_number = noticeOfWork.now_number;
+
+    let isPermitAmendmentTypeDropDownDisabled = true;
+    let permitAmendmentDropdown = this.props.permitAmendmentTypeDropDownOptions;
+    if (amendment && !isEmpty(amendment)) {
+      permitGenObject.permit_amendment_type_code = amendment.permit_amendment_type_code;
+
+      let amendmentType = PERMIT_AMENDMENT_TYPES.original;
+      const amendments = draftPermit.permit_amendments.filter(
+        (a) => a.permit_amendment_status_code !== "DFT"
+      );
+
+      if (amendments.length > 0) {
+        amendmentType =
+          amendments.filter(
+            (a) => a.permit_amendment_type_code === PERMIT_AMENDMENT_TYPES.amalgamated
+          ).length > 0
+            ? PERMIT_AMENDMENT_TYPES.amalgamated
+            : PERMIT_AMENDMENT_TYPES.amendment;
+      }
+
+      if (amendmentType === PERMIT_AMENDMENT_TYPES.amendment) {
+        permitAmendmentDropdown = this.props.permitAmendmentTypeDropDownOptions.filter(
+          (a) => a.value !== PERMIT_AMENDMENT_TYPES.original
+        );
+        isPermitAmendmentTypeDropDownDisabled = false;
+      }
+
+      this.setState({ permitAmendmentDropdown, isPermitAmendmentTypeDropDownDisabled });
+    }
 
     return permitGenObject;
   };
 
+  createPreviousAmendmentGenObject = (permit) => {
+    // gets and sorts in descending order amendments for selected permit
+    const amendments =
+      permit &&
+      permit.permit_amendments
+        .filter((a) => a.permit_amendment_status_code !== "DFT")
+        // eslint-disable-next-line no-nested-ternary
+        .sort((a, b) => (a.issue_date < b.issue_date ? 1 : b.issue_date < a.issue_date ? -1 : 0));
+    const previousAmendment = amendments[0];
+    previousAmendment.issue_date = formatDate(previousAmendment.issue_date);
+    previousAmendment.authorization_end_date = formatDate(previousAmendment.authorization_end_date);
+    return previousAmendment;
+  };
```

Comment:
Can you update to an object spread instead? It's more inline with our patterns.

```
 const previousAmendment ={
  ...amendments[0],
 issue_date: formatDate(previousAmendment.issue_date),
 authorization_end_date: formatDate(previousAmendment.authorization_end_date),
 }

Reply:
done